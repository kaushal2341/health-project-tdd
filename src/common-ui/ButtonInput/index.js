import React from 'react';
import {Button} from 'semantic-ui-react'
import Proptypes from 'prop-types'
export default class ButtonInput extends React.PureComponent {
onClickHandler = ()=>{
    this.props.onClickHandler();
}
render() {
    return(
        <>
        <Button
        id={this.props.id}
        type={this.props.type}
        active={this.props.active}
        animated={this.props.animated}
        as={this.props.as}
        attached={this.props.attached}
        basic={this.props.basic}
        circular={this.props.circular}
        className={this.props.className} 
        color={this.props.color}
        compact={this.props.compact}
        content={this.props.content}
        disabled={this.props.disabled}
        floated={this.props.floated}
        fluid={this.props.fluid}
        icon={this.props.icon}
        inverted={this.props.inverted}
        label={this.props.label}
        labelPosition={this.props.labelPosition}
        loading={this.props.loading}
        negative={this.props.negative}
        onClick={this.onClickHandler}
        positive={this.props.positive}
        primary={this.props.primary}
        role={this.props.role}
        secondary={this.props.secondary}
        size={this.props.size}
        tabIndex={this.props.tabIndex}
        toggle={this.props.toggle}
        >
        {this.props.buttonName}
        {this.props.children}
        </Button>
        </>
    )
}
}
React.propTypes ={
    id:Proptypes.string||Proptypes.number,
    type:Proptypes.string.isRequired,
    children:Proptypes.node,
    classes:Proptypes.object,
    onClick:Proptypes.func.isRequired,
    fullWidth:Proptypes.bool,
    size:Proptypes.string,
    disabled:Proptypes.bool,
    href:Proptypes.string,
    variant:Proptypes.variant,
    component:Proptypes.elementType,
    disableFocusRipple:Proptypes.bool,
    disableRipple:Proptypes.bool,
    color:Proptypes.string,
    buttonName:Proptypes.string.isRequired
}