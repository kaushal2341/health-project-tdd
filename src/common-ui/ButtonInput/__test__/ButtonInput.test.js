import React from 'react';
import ButtonInput from '..'
import { Button } from '@material-ui/core';

expect.addSnapshotSerializer(enzymeSerializer);
describe('Button Component', () => {
    let wrapper,
        onClickMock;
    beforeEach( () => {
        onClickMock = jest.fn()
        wrapper = mount(<ButtonInput type="submit" buttonName="Submit" onClickHandler={onClickMock}></ButtonInput>)
    });
    afterEach(()=>{
        wrapper.unmount()
    })
    
    it( 'should be defined' , () => {
       expect(wrapper).toBeDefined();
    });
    
    it( 'should have only one button' , () => {
        expect(wrapper.find('button')).toHaveLength(1)
    });
    
    it('should show name of the button',()=>{
        expect(wrapper.find('button').text()).toEqual(wrapper.prop('buttonName'));
    })

    it('should  have the exactly given props props',()=>{
        const propsToCheck= ['id','type','children','classes','onClick','fullWidth','size','disabled','href','variant','component','disableFocusRipple','disableRipple','color'];
        let buttonInputPropsKeys=Object.keys(wrapper.find(Button).props());
        propsToCheck.map((reqProp,i)=>{
            expect(reqProp).toContain(buttonInputPropsKeys[i]);
        })        
    });
    it('should only have type of submit or button', ()=>{
     const typeValue =['submit','button'];
     expect(typeValue).toContain(wrapper.find(Button).prop('type'))
    });
    describe('Button Input Simulation',()=>{
        let clickCallback;
        beforeEach(() => {
            clickCallback = jest.spyOn(wrapper.instance(),'onClickHandler');
            wrapper.instance().forceUpdate(); 
            wrapper.find(Button).simulate('click');
              
        });
        it('should call the onClick fuction handler when it is clicked',() => {
         expect(clickCallback).toHaveBeenCalled();
        });
        it('should call the props onClikHandler function after button is Clicked',() =>{
          expect(onClickMock).toHaveBeenCalled();
        })
    });
    describe('Snap Shot Testing',() => {
        it('should create or match previous snapshot of the component',() => {
            expect(wrapper).toMatchSnapshot(); 
        });
    })
})