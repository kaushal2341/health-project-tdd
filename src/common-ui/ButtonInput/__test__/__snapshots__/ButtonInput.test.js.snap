// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snap Shot Testing should create or match previous snapshot of the component 1`] = `
<ButtonInput
  buttonName="Submit"
  onClickHandler={[MockFunction]}
  type="submit"
>
  <Button
    as="button"
    onClick={[Function]}
    type="submit"
  >
    <Ref
      innerRef={
        Object {
          "current": <button
            class="ui button"
            type="submit"
          >
            Submit
          </button>,
        }
      }
    >
      <RefFindNode
        innerRef={
          Object {
            "current": <button
              class="ui button"
              type="submit"
            >
              Submit
            </button>,
          }
        }
      >
        <button
          className="ui button"
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>
      </RefFindNode>
    </Ref>
  </Button>
</ButtonInput>
`;
